sydent/replication/peer.py

```diff
@@ -83,8 +84,22 @@ def __init__(self, sydent, server_name, pubkeys):
         self.port = 1001
 
         # Get verify key for this peer
-        key_bytes = decode_base64(self.pubkeys[SIGNING_KEY_ALGORITHM])
-        self.verify_key = signedjson.key.decode_verify_key_bytes(SIGNING_KEY_ALGORITHM + ":", key_bytes)
+
+        # Check if their key is base64 or hex encoded
+        pubkey = self.pubkeys[SIGNING_KEY_ALGORITHM]
+        try:
+            # Check for hex encoding
+            pubkey_decoded = int(pubkey, 16)
```

Comment:
are you sure this does the right thing?
```
>>> pubkey_decoded = int('e2127f83a32cef77f315d5a2bf900ff959bf66357d0413294b3374a900ab8d1a', 16)
>>> signedjson.key.decode_verify_key_bytes("ed25519:", pubkey_decoded)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rav/work/synapse/env/local/lib/python2.7/site-packages/signedjson/key.py", line 100, in decode_verify_key_bytes
    key = nacl.signing.VerifyKey(key_bytes)
  File "/home/rav/work/synapse/env/local/lib/python2.7/site-packages/nacl/signing.py", line 68, in __init__
    raise exc.TypeError("VerifyKey must be created from 32 bytes")
nacl.exceptions.TypeError: VerifyKey must be created from 32 bytes
```

Reply:
Agh, yeah, the `int` is just a test, not the actual decoding step.